### PR TITLE
Update verbose logging documentation

### PR DESCRIPTION
**Pull request by Frassle**
Part fix for https://github.com/pulumi/pulumi/issues/10139

### CODE REVIEW

File: themes/default/content/docs/support/troubleshooting.md [hunk 1]
```diff
@@ -35,15 +35,17 @@ of issue reports.
...
 access.  Use the flag `--logflow` to apply the same log level to resource providers.
 
-> These logs may include sensitive information that is provided from your execution environment to
-your cloud provider (and which Pulumi may not even itself be aware of) so be careful to audit before
-sharing.
+> These logs may include sensitive information that is provided from your execution environment to your cloud
+provider (and which Pulumi may not even itself be aware of) so be careful to audit before sharing. At log
+levels below 11, Pulumi will avoid intentionally exposing any known credentials. At log level 11, Pulumi will
+intentionally expose some known credentials to aid with debugging, so these log levels should be used only
+when absolutely needed"?
```
Comment (aureq) [inline themes/default/content/docs/support/troubleshooting.md:48]: There's an extra `"?` at the end of the line. Is this intentional?

Comment (aureq) [reply themes/default/content/docs/support/troubleshooting.md:48]: Let me know of the following alternate message

```
Enabling verbose logging may reveal sensitive information (tokens, credentials...) that is provided from your execution environment to your cloud provider and which Pulumi may not even itself be aware of. It is recommended that you audit the saved logs and redact any information before sharing the logs.

At log level 10 or below, Pulumi will avoid intentionally exposing any known credentials. At log level 11, Pulumi will
intentionally expose some known credentials to aid with debugging, so these log levels should be used only
when absolutely needed.
```

**Review by aureq — COMMENTED** [review]
One very small comment that may need a change and a slightly different phrasing as a possible idea.

File: themes/default/content/docs/support/troubleshooting.md (hunk 1 above)
Comment (Frassle) [reply themes/default/content/docs/support/troubleshooting.md]: > There's an extra "? at the end of the line. Is this intentional?

Nope!

> Let me know of the following alternate message

That is all round better, so I'll go with that.

**Comment by aureq** [comment]
Looks good to me. I will let @cnunciato approuve the PR.

File: themes/default/content/docs/support/troubleshooting.md [hunk 2]
```diff
@@ -35,15 +35,18 @@ of issue reports.
...
 
-> These logs may include sensitive information that is provided from your execution environment to
-your cloud provider (and which Pulumi may not even itself be aware of) so be careful to audit before
-sharing.
+> Enabling verbose logging may reveal sensitive information (tokens, credentials...) that is provided from
+your execution environment to your cloud provider and which Pulumi may not even itself be aware of. It is
+recommended that you audit the saved logs and redact any information before sharing the logs. At log level 10
+or below, Pulumi will avoid intentionally exposing any *known* credentials. At log level 11, Pulumi will
+intentionally expose some known credentials to aid with debugging, so these log levels should be used only
+when absolutely needed.
```
Comment (cnunciato) [inline themes/default/content/docs/support/troubleshooting.md:49]: Looks good, but given the content here, I might suggest wrapping this in a warning callout instead:

```suggestion
{{% notes type="warning" %}}
Enabling verbose logging may reveal sensitive information (tokens, credentials...) that is provided from
your execution environment to your cloud provider and which Pulumi may not even itself be aware of. It is
recommended that you audit the saved logs and redact any information before sharing the logs. At log level 10
or below, Pulumi will avoid intentionally exposing any *known* credentials. At log level 11, Pulumi will
intentionally expose some known credentials to aid with debugging, so these log levels should be used only
when absolutely needed.
{{% /notes %}}
```

**Review by cnunciato — APPROVED** [review]
LGTM, but would suggest using a warning callout given the content of the message.

**Comment by Frassle** [comment]
@cnunciato are PR tests broken at the moment? Action failure doesn't look like it's due to this change.

**Pull request merged**